### PR TITLE
Add "recursive" attribute to git resource

### PR DESCRIPTION
**Pull request by mmasaki**
I propose an new attribute "recursive" into git resource.

Example:
```
git "foo" do
  repository "https://github.com/foo/including_submodules.git"
  recursive true
end
```

Then, "git clone --recursive https://github.com/foo/including_submodules.git" will be executed.

### CODE REVIEW

**Comment by ryotarai** [comment]
Great but I think the code is a little duplicated. How about the following?

```ruby
cmd = ['git', 'clone']
cmd << '--recursive' if attributes.recursive
cmd << attributes.repository << attributes.destination
run_command(cmd)
```

**Comment by mmasaki** [comment]
Tests were failed, but I think it's not this change's sake.
Please consider re-testing and merging this change.

File: lib/itamae/resource/git.rb [hunk 1]
```diff
@@ -29,7 +30,15 @@ def action_sync(options)
         if run_specinfra(:check_file_is_directory, attributes.destination)
           run_command_in_repo(['git', 'fetch', 'origin'])
         else
-          run_command(['git', 'clone', attributes.repository, attributes.destination])
+          if attributes.recursive
```
Comment (ryotarai) [inline lib/itamae/resource/git.rb]: L33-37: Please remove these lines

**Comment by mmasaki** [comment]
That's my mistake...

**Comment by ryotarai** [comment]
Thank you for your contribution!

**Pull request merged**